lib/storage/storage.go

```diff
@@ -165,6 +165,45 @@ type Storage struct {
 	isReadOnly atomic.Bool
 }
 
+// shouldDeleteMissingMetricID checks if metricID index entry is broken
+// and should be deleted by caller
+// There are the following expected cases when this may happen:
+//
+//  1. The corresponding metricID -> metricName/tsid entry isn't visible for search yet.
```

Comment:
Is it feasible at all to add a unit test these cases (not visible yet, unclean shutdown, snapshot)? If the time is the problem I guess it could be manipulated by letting the Storage accept a function that returns the current time and then overriding that function in unit tests (i.e. make it return whatever we want). 

As for unclean shutdown, not sure if this this possible to emulate in a unit test. Perhaps an integration test that starts a vm instance in a separate process and then kills that process?

Not for this PR. Just something to think about.

Reply:
I was thinking about such tests, but it seems to flaky for me. And it also depends on knowledge of lib/mergeset internals. Which looks too much for unit test at lib/storage. 

 For good test, we need to :
1) route metricID  index records to the specific rawItemShard at lib/mergeset
2) flush only this shard to the disk
3) perform query and observe missing records
4) flush left rawItemShards 
5) ensure that all index records was found